core/trino-main/src/test/java/io/trino/memory/TestTotalReservationOnBlockedNodesLowMemoryKiller.java

```diff
@@ -75,4 +78,58 @@ public void testSkewedQuery()
                         toNodeMemoryInfoList(memoryPool, queries)),
                 Optional.of(KillTarget.wholeQuery(new QueryId("q_1"))));
     }
+
+    @Test
+    public void testPreferKillingTasks()
+    {
+        int memoryPool = 12;
+        Map<String, Map<String, Long>> queries = ImmutableMap.<String, Map<String, Long>>builder()
+                .put("q_1", ImmutableMap.of("n1", 0L, "n2", 8L, "n3", 0L, "n4", 0L, "n5", 0L))
+                .put("q_2", ImmutableMap.of("n1", 3L, "n2", 5L, "n3", 2L, "n4", 4L, "n5", 0L))
+                .put("q_3", ImmutableMap.of("n1", 0L, "n2", 0L, "n3", 11L, "n4", 0L, "n5", 0L))
+                .buildOrThrow();
+
+        Map<String, Map<String, Map<String, Long>>> tasks = ImmutableMap.<String, Map<String, Map<String, Long>>>builder()
+                .put("q_2", ImmutableMap.of(
+                        "n1", ImmutableMap.of("t1", 1L, "t2", 3L),
+                        "n2", ImmutableMap.of("t3", 3L, "t4", 1L, "t5", 1L),
```

Comment:
The total task reservation of q_2 on n2 doesn't equal to the query reservation of q_2 on n2, it it expected? I understand that it doesn't affect the effectiveness of the test, though.

Reply:
Actually, it matches: 3+1+1 = 5. But yeah - the killing logic does not care about total query reservation for queries with task level retries.